packages/engine/src/datastore/schema/field_spec/mod.rs

```diff
@@ -104,17 +104,26 @@ impl FieldSource {
 pub struct FieldKey(String);
 
 impl FieldKey {
+    /// Returns the key as string
     pub fn value(&self) -> &str {
         &self.0
     }
 
-    pub(in super::super) fn from_key_as_str(key_as_str: &str) -> Self {
-        Self(key_as_str.to_string())
+    /// Returns a string as key
+    pub(in super::super) fn new(key: &str) -> Self {
```

Comment:
I don't feel strongly against this, I know in general a `new` method is more conventional here. But I think `from_key_as_str` (or something similar) was more explicit that you're getting a fully-qualified key as the input, not a field name.

Reply:
Hmm, what about `from_key_str`?

Reply:
I'd prefer `new` over that. I think `key_str` isn't particularly clear. At least the previous one was clear it was a key `as` a str